src/shared/filters/all-exceptions.filter.ts

```diff
@@ -1,20 +1,20 @@
 import {
   ArgumentsHost,
   Catch,
-  ExceptionFilter,
   HttpException,
   HttpStatus,
 } from '@nestjs/common';
-import { Request, Response } from 'express';
 import { ConfigService } from '@nestjs/config';
+import { GqlArgumentsHost, GqlExceptionFilter } from '@nestjs/graphql';
+import { Request, Response } from 'express';
 
 import { REQUEST_ID_TOKEN_HEADER } from '../constants';
-import { AppLogger } from '../logger/logger.service';
 import { BaseApiError } from '../errors/base-api-error';
+import { AppLogger } from '../logger/logger.service';
 import { createRequestContext } from '../request-context/util';
 
 @Catch()
-export class AllExceptionsFilter<T> implements ExceptionFilter {
+export class AllExceptionsFilter<T> implements GqlExceptionFilter {
```

Comment:
If the project also has some REST endpoints, do we need to consider that case as well?

Reply:
@FelipeCO14 Added support with the following if condition:
```
if (host.getType() == 'http') {
      ctx = host.switchToHttp();
      req = ctx.getRequest();
      res = ctx.getResponse();
    } else {
      ctx = GqlArgumentsHost.create(host).getContext();
      req = <Request>ctx.req;
      res = <Response>ctx.req.res;
    }
```

Here is how it looks when the filter catches HTTP rest exceptions:
```
{
    "error": {
        "statusCode": 500,
        "message": "Fake forced exception from REST controller.",
        "errorName": "InternalServerErrorException",
        "details": {
            "statusCode": 500,
            "message": "Fake forced exception from REST controller.",
            "error": "Internal Server Error"
        },
        "requestId": "0f415994-8d52-4b55-a4e6-01043224a6ad",
        "timestamp": "2022-01-25T10:37:27.298Z"
    }
}
```